### PR TITLE
Create `CapsuleWithWhiteText` as `ButtonStyle`

### LINKED ISSUE #129: Button(s) modifiers.

### PR DESCRIPTION
**Pull request by marinkman**
## Summary
This PR is a part of the work to resolve #129 

## UI Preview (Screenshots)
<details>
  <summary>iPhone 14 Pro Max - 6.7"</summary>

Screenshot 2023-06-09 at 12 40 40

</details>
<details>

  <summary>iPhone 14 Pro -  6.1"</summary>

Screenshot 2023-06-09 at 12 40 40

</details>
<details>

  <summary>iPhone SE 3rd gen - 4.7"</summary>

Screenshot 2023-06-09 at 12 42 30

</details>

## Changes
- Created `ButtonStyle` folder
- Created `CapsuleWithWhiteText` struct
- Applied `CapsuleWithWhiteText` in the `OwnerProfileView`

## Checks to complete
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] PR's title has a conventional commit style (short and descriptive).
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+/// A type that applies standard interaction behavior and a custom appearance to all buttons within a view hierarchy.
+/// The button will have a capsule shape of the specified color.
+/// The button will have a label with the white center-aligned text.
+/// The button size will adjust to the length of the text.
+///
+/// To configure the current button style for a view hierarchy, use the `buttonStyle(_:)` modifier.
+struct CapsuleWithWhiteText: ButtonStyle {
+    /// A capsule color.
+    let capsuleColor: Color
```
Comment (dersim-davaod) [inline Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift:11]: nit:

```suggestion
    let backgroundColor: Color
```

?

**Review by dersim-davaod — COMMENTED** [review]
Please consider few improvements:

File: Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import SwiftUI
+
+/// A type that applies standard interaction behavior and a custom appearance to all buttons within a view hierarchy.
```
Comment (iBYarchitect) [inline Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift:3]: typo
```suggestion
/// A type that applies standard interaction behaviour and a custom appearance to all buttons within a view hierarchy.
```

**Review by iBYarchitect — CHANGES_REQUESTED** [review]
Found one typo

File: Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift (hunk 2 above)
Comment (marinkman) [reply Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift:3]: There is no typo here. Both options are possible.

File: Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift (hunk 1 above)
Comment (marinkman) [reply Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift]: Fixed.

File: Woof/Woof/ViewModel/OwnerProfileViewModel.swift [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import Foundation
+import SwiftUI
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/ViewModel/OwnerProfileViewModel.swift:2]: We do not need import `Foundation`, if we import `SwiftUI`
```suggestion
import SwiftUI
```

**Review by NadzeyaShpakouskaya — COMMENTED** [review]
Well done, @marinkman  🎊 
I have one suggestion if it makes sense:

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 4]
```diff
@@ -30,20 +30,17 @@ struct OwnerProfileView: View {
...
-                    .background(
-                        Capsule()
-                            .foregroundColor(!viewModel.name.isEmpty ? Color.App.purpleDark : Color.App.grayDark)
-                    )
-                    .foregroundColor(Color.App.white)
-            })
+            }
+            .buttonStyle(
+                CapsuleWithWhiteText(
+                    capsuleColor: !viewModel.name.isEmpty ? Color.App.purpleDark : Color.App.grayDark
```
[the remark this answers is not in the document]
Comment (marinkman) [reply Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift]: Done.

File: Woof/Woof/ViewModel/OwnerProfileViewModel.swift (hunk 3 above)
Comment (marinkman) [reply Woof/Woof/ViewModel/OwnerProfileViewModel.swift]: Fixed.

**Review by julyakmita — APPROVED** [review]
Good job!)
LGTM.

**Review by NadzeyaShpakouskaya — APPROVED** [review]
LGTM

File: Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+/// The button will have a capsule shape.
+/// The color of the button will depend on whether it is enabled.
+/// The button will be dark purple color if it is enabled, otherwise dark gray.
+/// The button will have a label with the white center-aligned text.
+/// The button size will adjust to the length of the text.
+///
+/// To configure the current button style for a view hierarchy, use the `buttonStyle(_:)` modifier.
+struct CapsuleWithWhiteText: ButtonStyle {
+    @Environment(\.isEnabled) var isEnabled
```
Comment (kirpichonok) [inline Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift:12]:
```suggestion
    @Environment(\.isEnabled) private var isEnabled
```

**Review by iBYarchitect — APPROVED** [review]
LGTM

**Review by kirpichonok — CHANGES_REQUESTED** [review]
Good job, but I have one suggestion.

File: Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift (hunk 5 above)
Comment (marinkman) [reply Woof/Woof/View/ReusableViews/ButtonStyle/CapsuleWithWhiteText.swift]: Done.

**Comment by dersim-davaod** [comment]
Please give the PR proper title.

**Review by kirpichonok — APPROVED** [review]
LGTM

**Review by dersim-davaod — APPROVED** [review]
LGTM

**Pull request merged**